Intersect.Client/Entities/Player.cs

```diff
@@ -1237,30 +1237,58 @@ private void ClearTarget()
             }
         }
 
-        public bool TryPickupItem()
+        /// <summary>
+        /// Attempts to pick up an item at the specified location.
+        /// </summary>
+        /// <param name="x">The X location on the current map.</param>
+        /// <param name="y">The Y location on the current map.</param>
+        /// <param name="uniqueId">The Unique Id of the specific item we want to pick up, leave <see cref="Guid.Empty"/> to not specificy an item and pick up the first thing we can find.</param>
+        /// <param name="firstOnly">Defines whether we only want to pick up the first item we can find when true, or all items when false.</param>
+        /// <returns></returns>
+        public bool TryPickupItem(int x, int y, Guid uniqueId = new Guid(), bool firstOnly = false)
         {
             var map = MapInstance.Get(CurrentMap);
             if (map == null)
             {
                 return false;
             }
 
-            foreach (var item in map.MapItems)
+            var location = new Point(x, y);
+            if (!map.MapItems.ContainsKey(location) && map.MapItems[location].Count < 1)
```

Comment:
Should be || not &&

Reply:
Is that tested? Maybe my demented mind might be wrong but that sounds like it might cause an NRE.

I'll have to look into that when I get the time.

Reply:
I am dumb, and should not say things at 6 in the morning.